### PR TITLE
Update datafusion protobuf definitions

### PR DESCRIPTION
**Pull request by palaska**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

Keep things in sync with the currently used Datafusion version.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by andygrove — APPROVED** [review]
Thanks @palaska

File: dev/docker/ballista-builder.Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM rust:1.74.1-buster
+FROM rust:1.81-bullseye
```
Comment (palaska) [inline dev/docker/ballista-builder.Dockerfile:18]: `buster`'s package repo did not have a high enough version of `protoc` that supports `experimental_allow_proto3_optional` flag, that's why I bumped

File: ballista/core/build.rs [hunk 2]
```diff
@@ -37,10 +37,13 @@ fn main() -> Result<(), String> {
     // We don't include the proto files in releases so that downstreams
     // do not need to have PROTOC included
     if Path::new("proto/datafusion.proto").exists() {
+        println!("cargo:rerun-if-changed=proto/datafusion_common.proto");
         println!("cargo:rerun-if-changed=proto/datafusion.proto");
         println!("cargo:rerun-if-changed=proto/ballista.proto");
         tonic_build::configure()
+            .extern_path(".datafusion_common", "::datafusion_proto_common")
             .extern_path(".datafusion", "::datafusion_proto::protobuf")
+            .protoc_arg("--experimental_allow_proto3_optional")
```
Comment (palaska) [inline ballista/core/build.rs:46]: this is the flag, same seems to be used in Datafusion as well

**Comment by palaska** [comment]
> Thanks @palaska

No worries @andygrove, made couple small changes. It should be good to go now

**Pull request merged**